lib/suspenders/cleanup/organize_gemfile.rb

```diff
@@ -0,0 +1,119 @@
+module Suspenders
+  module Cleanup
+    class OrganizeGemfile
+      def self.perform(gemfile)
+        new(gemfile).perform
+      end
+
+      attr_reader :gemfile, :current_lines, :new_lines, :new_lines_markers,
+        :current_group, :gem_groups
+
+      def initialize(gemfile)
+        @gemfile = gemfile
+
+        @current_lines = File.read(gemfile).lines
+        @new_lines = []
+        @new_lines_markers = []
+
+        @current_group = nil
+        @gem_groups = {}
+      end
+
+      def perform
+        # Remove line breaks to ensure consistent spacing
```

Comment:
What do you think of something like:

```ruby
def perform
  group_gems
  add_consolidated_gem_groups
  add_relevant_newlines
  final_cleanup
end
```

This way we get a high-level view of what it means to organize a Gemfile.

Reply:
Addressed here: fd6db90f91ff08fe849528c35482afbc0f71c5f1